source/common/http/header_map_impl.cc

```diff
@@ -464,6 +464,17 @@ void HeaderMapImpl::remove(const LowerCaseString& key) {
   }
 }
 
+void HeaderMapImpl::remove(const std::regex& regex) {
+  for (auto i = headers_.begin(); i != headers_.end();) {
+    absl::string_view key = i->key().getStringView();
+    if (std::regex_search(key.begin(), key.end(), regex)) {
```

Comment:
consider std::list::remove_if ?

Reply:
Hah - just converted over and got Matt's comment about inline headers.  I think the additional complexity to remove the constant time references may make this version preferable.  Will look though!